python/tests/test_tc.py

```diff
@@ -0,0 +1,420 @@
+# Copyright (c) 2017-present, Facebook, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##############################################################################
+import unittest
+
+import tempfile
+
+import torch
+import torch.cuda
+
+import tensor_comprehensions as tc
+
+tuner_config = tc.TunerConfig(threads=8, generations=3, pop_size=8)
+
+class TestTC(unittest.TestCase):
+    #
+    # Self explicit
+    #
+    def test_imports(self):
+        from tensor_comprehensions.tclib import logtostderr
+        from tensor_comprehensions.tclib import debug_lang
+        from tensor_comprehensions.tclib import debug_halide
+        from tensor_comprehensions.tclib import debug_tc_mapper
+        from tensor_comprehensions.tclib import debug_tuner
+        from tensor_comprehensions.tclib import dump_cuda
+
+        from tensor_comprehensions.tclib import CompilationCache
+        from tensor_comprehensions.tclib import MappingOptions
+        from tensor_comprehensions.tclib import MappingOptionsCache
+        from tensor_comprehensions.tclib import Tuner
+        from tensor_comprehensions.tclib import TunerConfig
+
+    #
+    # Construct a MappingOptions object programmatically from Python-land
+    #
+    def test_mapping_options(self):
+        options = (
+            tc.MappingOptions('naive')
+            .useSharedMemory(True)
+            .unrollCopyShared(False)
+            .mapToBlocks([256, 8])
+            .mapToThreads([4, 16, 4])
+            .tile([2, 8, 64, 128])
+            .unroll(128)
+            .fixParametersBeforeScheduling(False)
+            .scheduleFusionStrategy("Max")
+            .outerScheduleFusionStrategy("Preserve3Coincident"))
+
+    #
+    # Simple TC example with explicit 'naive' compilation
+    #
+    def test_tc(self):
+        add = tc.define(
+            "def add(float(N) A, float(N) B) -> (C) { C(i) = A(i) + B(i) }",
+            "add")
+        A, B = torch.randn(100, device='cuda'), torch.randn(100, device='cuda')
+        add.compile(A, B, mapping_options='naive')
+        C, = add(A, B)
```

Comment:
Something we also did in PyTorch is to automatically treat single returns as values, and multi returns as tuples. This is more consistent with Python (`return x` is of `type(x)`, `return x, y` is a `tuple`).